### PR TITLE
Testing the action on GitHub-hosted ubuntu runners

### LINKED ISSUE #38: Test action performance on GitHub-hosted runners
## Summary 

The current recommendation for running this action on Ubuntu is a docker container instead of GitHub-hosted runners. It would be a good idea to test the working of this action on GitHub-hosted runners as many workflows out there use them for their workflows.

Additionally, add documentation about this including a point on `ubuntu-24.04` being a beta version as specified [here](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories).

### PR DESCRIPTION
**Pull request by sauk2**
Closes #38 

## Summary

The PR adds jobs to `test.yml` to run the action on GitHub ubuntu runners. 

Updated the README with instructions. Also added a note about the `ubuntu-24.04` being a beta version.

### CODE REVIEW

**Comment by sauk2** [comment]
@j-rivero I have not changed the workflow for the pre-release and nightly jobs. These workflows still run inside a docker container. Do you want me make similar changes there too?

**Comment by j-rivero** [comment]
> @j-rivero I have not changed the workflow for the pre-release and nightly jobs. Those workflows still run inside a docker container. Do you want me to make similar changes there too?

Thanks for the PR ! This confirms that we don't need to modify any single line of code of the setup-gazebo (the typescript and command instructions). So only the consumer of that package (the yaml code of the user of setup-gazebo, in this case test.yml)  need to specify if it want to use containers or not.

Said this, I think that would be a good idea to have in the README:
  * As the very basic use case: no containers at all
  * After this, show how can someone run it in a container (like we currently have it)

Another question is: how do we want to create our tests, that is actually a consumer of the setup-gazebo action. Given that there is no benefit for us on having the tests running on the bare metal runner, it is probably a good idea to the main tests (basic, nightly and prerelease) keep them running on containers and maybe add a test that runs on bare-metal to be sure that there is nothing in that use case that is broken.

So summarizing:

  * I would not change the `test.yml` code just add a new test running on bare metal.
  * Change README so the first use case for the user is the simple use of the action on bare-metal runners. Keeping the current use case below as an alternative for them to use containers.

Sounds good to you @sauk2 ?

**Comment by sauk2** [comment]
> Sounds good to you @sauk2 ?

Thanks. Sounds great!!

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -108,13 +197,29 @@ jobs:
...
+          node-version: '20.x'
+      - name: 'Check Gazebo installation on Ubuntu runner'
+        uses: ./
+        with:
+          required-gazebo-distributions: 'ionic'
+          use-gazebo-nightly: 'true'
+      - name: 'Test Gazebo installation'
+        run: 'gz sim --versions'
+
+  test_gazebo_install_ubuntu_nightly_docker:
```
Comment (j-rivero) [inline .github/workflows/test.yml:215]: We can leave the version of running the pre-release and nightly tests on docker containers and remove the github runners. Do not need to test it twice in different environments, looks to me like too much testing that does not differ too much.

File: README.md [hunk 2]
```diff
@@ -82,14 +113,14 @@ The following code snippet shows the installation of Gazebo Harmonic on Ubuntu N
 
 #### Iterating on all Gazebo and Ubuntu combinations
 
-This workflow shows how to spawn one job per Gazebo release. It iterates over all specified Gazebo and Ubuntu combinations using Docker. For example, Gazebo Garden is paired with Ubuntu Focal while Gazebo Harmonic is paired with Ubuntu Jammy.
+This workflow shows how to spawn one job per Gazebo release and iterates over all specified Gazebo and Ubuntu combinations. It is done by defining a `matrix` to iterate over jobs.
+
+- *Using GitHub-hosted runners*
```
Comment (j-rivero) [inline README.md:118]:
```suggestion
- *Default: using GitHub-hosted runners systems*
```

File: README.md [hunk 3]
```diff
@@ -135,10 +166,86 @@ This workflow shows how to spawn one job per Gazebo release. It iterates over al
...
+            uses: gazebo-tooling/setup-gazebo@<full_commit_hash>
+            with:
+              required-gazebo-distributions: 'harmonic'
+              use-gazebo-prerelease: 'true'
+              use-gazebo-nightly: 'true'
+          - name: 'Test Gazebo installation'
+            run: 'gz sim --versions'
+```
+
+- *Using Ubuntu docker containers*
```
Comment (j-rivero) [inline README.md:247]: I think that we can remove the option of the docker containers for prerelease and nightlies in favour of reducing the README lenght.

**Review by j-rivero — COMMENTED** [review]
Minor changes suggested and we are good to go :rocket:

**Comment by sauk2** [comment]
> Minor changes suggested and we are good to go 🚀

Thanks for your feedback! I have made the changes you suggested. All check pass, ready for merge :rocket:

**Review by j-rivero — APPROVED** [review]
Yuhuuuuuu

**Pull request merged**